src/plugins/shutdown_button/ShutdownButton.hh

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#ifndef IGNITION_GUI_PLUGINS_SHUTDOWNBUTTON_HH_
+#define IGNITION_GUI_PLUGINS_SHUTDOWNBUTTON_HH_
+
+#include <memory>
+
+#ifdef _MSC_VER
+#pragma warning(push, 0)
+#endif
+#include <ignition/msgs.hh>
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif
+#include <ignition/transport.hh>
```

Comment:
These includes can be moved to the cc file. And ideally we'd include just what we need (i.e. the `pb.h` include and `Node.hh`)

Reply:
ShutdownButton is now just a thin plugin calling the window close event. So most includes went away. Some functionality moved to MainWindow, and I made the includes as precise as possible.